orttraining/orttraining/core/graph/optimizer_config.h

```diff
@@ -62,6 +62,7 @@ struct OptimizerGraphConfig {
   std::string loss_scale_input_name{};  // empty string means no loss scaling factor is applied
   AdasumReductionType adasum_reduction_type{AdasumReductionType::None};
   bool enable_grad_norm_clip{true};
+  ONNX_NAMESPACE::TensorProto_DataType fp16_type{ONNX_NAMESPACE::TensorProto_DataType_FLOAT16};
```

Comment:
>ONNX_NAMESPACE::TensorProto_DataType [](start = 2, length = 36)

I left the same comment in Vincent's PR

I feel it's better to introduce an enum type for this purpose... 

ONNX_NAMESPACE::TensorProto_DataType has too many type, such as int and bool, which is not a meaningful value here.

Reply:
Maybe I can make the change after this PR? I feel like in this PR we still use the name "float16"/"fp16" for both fp16 and bf16 case, which is confusing. I tried to rename most of them in my PR.

Reply:
thanks Vincent. yes i agree on use an enum type, we might later have some special type that is not in onnx namespace yet...